_data/accessibility-tests/search.yml

```diff
@@ -0,0 +1,165 @@
+    test_type: keyboard
+    version_tested: 3.8.0
+    wcag_criterion: 1.3.2
+  - summary: |
+      The text field and the search button each have a 3:1 contrast outline against the background. 
+      The outline size is greater than or equal to the perimeter of the search field. 
+    summary_additional: When you use a keyboard to tab into a button or text field, you see an indicator and thick outline. 
+    test_status: pass
+    test_type: keyboard
+    version_tested: 3.8.0
+    wcag_criterion: 2.4.7
+  - summary: The focus indicator has proper size, perimeter and contrast.
+    summary_additional: |
+      When you use a keyboard to tab into the search component, the focus indicator is clearly visible with the 
+      proper size, perimeter, and contrast.
+    test_status: pass
+    test_type: keyboard
+    version_tested: 3.8.0
+    wcag_criterion: 2.4.13
+# Screen reader tests
+  - summary: Input search field announces purpose.
+    summary_additional: |
+      When you tab into the text input search field, you hear "search edit," so you know it's a search field you can type into.
+    test_status: pass
+    test_type: screen_reader
+    version_tested: 3.8.0
+    wcag_criterion: 1.3.5
+  - summary: Icon and button announce search upon initial interaction.
+    summary_additional: |
+      When tabbing onto the search icon and button with your screen reader, you hear the words "search button."
+    test_status: pass
+    test_type: screen_reader
+    version_tested: 3.8.0
+    wcag_criterion: 1.3.6
+  - summary: Screen reader announces elements to match visual order on screen. 
+    summary_additional: |
+      When using a screen reader, the search field and button are announced in the same order that they appear on the page.
+    test_status: conditional
+    test_type: screen_reader
+    version_tested: 3.8.0
+    wcag_criterion: 1.3.2
+  - summary: You hear the roles "button" and "text edit."
+    summary_additional: |
+      When you use a screen reader to search and you tab to the search button, you hear the role "button." 
+      When you tab to the text input field, you hear "text edit," which indicates the role of the field.
+    test_status: pass
+    test_type: screen_reader
+    version_tested: 3.8.0
+    wcag_criterion: 4.1.2
+  - summary: You hear the labels for "search" and "search button."
+    summary_additional: |
+      When you use a screen reader to search, the label for the text field says "search." 
+      The button label is called "search button."
+    test_status: pass
+    test_type: screen_reader
+    version_tested: 3.8.0
+    wcag_criterion: 2.4.6
+  - summary: Screen reader announces that search results will be displayed.
+    summary_additional: |
+      When you complete a search using a screen reader, you are informed of where search results will appear. 
+      You may be taken to a results page, or you may be notified that search results will display on the same page as the search field.
```

Comment:
@amycole501 @alex-hull 
As a dev implementing a search component, I would be curious how this should be communicated. Is it just adding detail to an aria-label on the search button? How have you seen this done in other implementations?Wondering if we give some clues to how a user can help this test pass.

Reply:
I think if it were something that described (specific input type?) the action prior to someone clicking the search button, that would be sufficient for this criteria. Like if the search button said something like, "complete search" or "search" before someone actually triggered the action. Once some interacts with the submit button, the person should hear that they have landed on a search results page.

Reply:
Site searches are ubiquitous and common enough that you may not have to warn someone of how or where the results will display (unless the results open in a new window).

Alex is correct, the search button does need to be specific and not just say "button" (for example).

Finally, this criteria may ONLY need to apply when someone has more than one search element on a page. 

For example, a site search in the top right may not need this check. BUT a database search which is separate from a sitewide search needs to include instructions on where and how the results will display. 

![image](https://github.com/uswds/uswds-site/assets/117778073/a2c477c0-4e89-4d6b-b932-ba9869201f64)

Reply:
I'm curious if the search result locations should be on the search input since some users may hit `enter` and never tab to the button itself.

It sounds like the correct implementation would be adding the intended search location after the search label. Something like:

`Search the glossary. Results will update in the following list` or `Results will be opened in a new page`

Does this sound like it's on the right track to meeting this criteria?

Reply:
Ooh, good observation. Yes, I think those would be appropriate implementations. And I think this would be a useful implementation for those who even tab to submit their search results. Maybe one of the above examples could be stated in the check to help with Amy L's question on providing hints for passing.

Reply:
This one is a bit tricky to interpret. Looking deeper at the [Understanding 3.2.2 criterion](https://www.w3.org/WAI/WCAG21/Understanding/on-input.html), it seems like this is mainly inputs that would not _normally_ change the content on the page, but not necessarily for submission buttons.

> Changing the setting of any user interface component is changing some aspect in the control that will persist when the user is no longer interacting with it. So checking a checkbox, entering text into a text field, or changing the selected option in a list control changes its setting, but activating a link or a button does not.
...
This Success Criterion covers changes in context due to changing the setting of a control. Clicking on links or tabs in a tab control is activating the control, not changing the setting of that control.

It sounds to me like this essentially means the page/form context does not change before it is expected. For example, the form changes when a radio or checkbox is selected without submitting the changes, but button / link clicks are considered _expected_ changes. 

It also [advises](https://arc.net/l/quote/icevonbd) to warn users when results will be opened in a new tab but it is not required for this criterion.

With this new context, do you think we should rewrite the criterion to be something along the lines of:

> **Page context does not change before search submission.**
> Updating page context or search results before expected can cause confusion for users of assistive technology. If search results are opened within a new window, advanced warning should be given alongside search instructions in the `aria-label` or button text.

Reply:
Also noting `3.2.2` would be a good addition to the radio and checkbox component checklists

Reply:
Updated in [aa5c9ea](https://github.com/uswds/uswds-site/pull/2694/commits/aa5c9ea518ba34c2803123d97ee809543f7ac5cd)

I left it in the screen reader section. Let me know if you think this needs to be moved to general instead!

Reply:
Discussed internally, moved the test to the `General` section

Resolved in [ffde89f](https://github.com/uswds/uswds-site/pull/2694/commits/ffde89f1ab108643199845affde224c8bf01d9f9)

Reply:
Marking this as resolved because the initial comment has been addressed. Continued the conversation about the updated text in this [comment thread](https://github.com/uswds/uswds-site/pull/2694#discussion_r1636739897).